compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1004,10 +1004,13 @@ pub enum Rvalue<'tcx> {
     /// This is the cause of a bug in the case where the repetition count is zero because the value
     /// is not dropped, see [#74836].
     ///
+    /// The third param is true if the value we're repeating is an enum variant
+    /// with no fields so we can emit a memset.
+    ///
     /// Corresponds to source code like `[x; 32]`.
     ///
     /// [#74836]: https://github.com/rust-lang/rust/issues/74836
-    Repeat(Operand<'tcx>, ty::Const<'tcx>),
+    Repeat(Operand<'tcx>, ty::Const<'tcx>, bool),
```

Comment:
It feels weird to need *this* specific property on `RValue`?  It feels like it could be gotten from looking at the type of the operand when needed.  It doesn't seem like it ought to be part of MIR syntax.

(Also, primitives in tuple-variants are often a bad idea, since there's no type hint for what it means.)

Reply:
I put the property into the rvalue since the actual variant being written is lost at the point I use the information - we only have a value loaded from the generated temporary (i.e. `(i8:  %9 = zext i1 %5 to i8)`)

Reply:
If this is known at MIR building but not later, one possibility might be to add a [`NonDivergingIntrinsic::MemSet`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/enum.StatementKind.html#variant.Intrinsic) that you emit instead of the `Repeat` sometimes.

(That's still an addition to the language that would need a spec, so it's still better to avoid if possible, but it's much lower impact than changing a top-level statement.)